src/hooks/useNavigation.ts

```diff
@@ -0,0 +1,26 @@
+import { useRouter } from 'next/router';
```

Comment:
This hook meant to centerlize the user url building so we won't put `/u/${userId}` everywhere.
We will also need the `getPreviousRoute` and / or keeping navigation state for [Filter params are not preserved when navigating to a user page](https://github.com/orgs/Coding-Coach/projects/3#card-78129859)

Reply:
I absolutely love the idea of centralizing the concerns of URL construction.  Although, I think we should separate the concerns of building the URLs from the concerns of navigating to the URLs.  Meaning that this hook would only be responsible for things like `getUserRoute`, but we'd leave the routing to the `<Link>` component instead of using `router.push`.  I suggest this because there's a lot of advantages to using the `<Link>` component.  Things like prefetching, accessibility, and SEO.

If we did that, it wouldn't be necessary to use a hook at all for route building.  We could, instead, use a utility file.  The advantage of a utility file is that it sends a clear message (IMO) that this files entire job is URL construction; nothing more.

Whether you choose the utility file option or the hook option, how do you feel about rearranging the way the methods are organized?  Instead of a single, top-level function such as `getUserRoute` we could use an object with nested objects that mimic our nested routes.  Here's an example:

``` javascript
export const useNavigation = () => {
  const urls = {
    user: {
      get: (user: User) => `/u/${user._id}`;
    }
  }
  
  return urls
};
```

The benefits of this approach may not be immediately obvious, but they really shine through when nested routes are added.

``` javascript
export const useNavigation = () => {
  const urls = {
    user: {
      get: (user: User) => `/u/${user._id}`;
    },
    me: {
      get: () => '/me',
      requests: {
        get: () => '/requests',
      },
      admin: {
        get: () => '/admin',
      }
    }
  }
  
  return urls
};
```

This is all personal preference and I'm _very_ OK rolling with what you have already.  I wanted to share a pattern I've been working with for a while to see if you like it, but it's not worth much effort to convert.

Reply:
I love that idea and implemented it. Let me know what you think